pkg/service/update/update.go

```diff
@@ -0,0 +1,141 @@
+package updateBinary
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/charmbracelet/glamour"
+	"go.keploy.io/server/utils"
+	"go.uber.org/zap"
+)
+
+// GitHubRelease holds information about the GitHub release.
+type GitHubRelease struct {
+	TagName string `json:"tag_name"`
+	Body    string `json:"body"`
+}
+
+// NewUpdater initializes a new updater instance.
+func NewUpdater(logger *zap.Logger) Updater {
+	return &updater{
+		logger: logger,
+	}
+}
+
+// Updater defines the contract for updating keploy.
+type Updater interface {
+	Update()
+}
+
+var ErrGitHubAPIUnresponsive = errors.New("GitHub API is unresponsive")
+
+// Update initiates the update process for the Keploy binary file.
+func (u *updater) Update() {
+	currentVersion := utils.Version
+
+	isDockerCmd := len(os.Getenv("IS_DOCKER_CMD")) > 0
```

Comment:
Handle "-dev" scenario here.

Reply:
Do we not want to allow the use of update command on -dev version? Could you kindly elaborate?

Reply:
I have added the code to skip the update if it is on "-dev" version, let me know if that is the intended behaviour

Reply:
yes because developer doesn't use hosted binaries